cdap-docs/developers-manual/source/security/authorization-extensions.rst

```diff
@@ -8,4 +8,42 @@
 Authorization Extensions
 ========================
 
-Authorization backends in CDAP are implemented as extensions.
+Authorization backends in CDAP are implemented as extensions. Authorization extensions are loaded and executed in their
+own Java classloader. The reason for implementing them as extensions is so that a variety of existing authorization
+systems can be plugged-in to CDAP, and can execute using their own classloader without having to worry about conflicts
+with CDAP's system classloader. Each authorization extension is designed to be a fully self-contained JAR file,
+packaged with the required versions of all its dependencies. This is similar to how CDAP applications
+are packaged. In fact, users can use the CDAP application archetype to write their own authorization extension for
```

Comment:
Just curious have we tested using cdap application archetype for developing extensions ? We put app.main.class and some other property in the pom or the maven project which changes how the MANIFEST file is generated. Will that affect how extension jar is loaded into cdap ?